config/prometheus/monitor.yaml

```diff
@@ -14,7 +14,11 @@ spec:
       port: https
       scheme: https
       tlsConfig:
-        caFile: /etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt
+        ca: 
+          secret:
+            key: 'tls.crt'
+            name: vault-config-operator-certs
+            optional: false
```

Comment:
@raffaelespazzoli for some reason I didn't need to give the prometheus serviceaccount permission (update the Role in the rbac config section) to read Secrets in the namespace. Tested successfully using kind and in OCP after a bundle deploy.

Reply:
Sounds like the operator has access to the Secrets, prometheus itself doesn't need rbac to Secrets.